include/sys/zfs_ioctl.h

```diff
@@ -70,10 +70,36 @@ extern "C" {
 #define	ZFS_ACLTYPE_POSIX		1
 #define	ZFS_ACLTYPE_NFSV4		2
 
+/*
+ * The drr_versioninfo field of the dmu_replay_record has the
+ * following layout:
+ *
+ *	64	56	48	40	32	24	16	8	0
+ *	+-------+-------+-------+-------+-------+-------+-------+-------+
+ *	|reserve|		  feature-flags			    |C|S|
+ *	+-------+-------+-------+-------+-------+-------+-------+-------+
+ *
+ * The low order two bits indicate the header type: SUBSTREAM (0x1)
+ * or COMPOUNDSTREAM (0x2).  Using two bits for this is historical:
+ * this field used to be a version number, where the two version types
+ * were 1 and 2.  Using two bits for this allows earlier versions of
+ * the code to be able to recognize send streams that don't use any
+ * of the features indicated by feature flags.
+ *
+ * The top 8 bits are reserved for future expansion. At time of writing there
+ * are no plans for these. If you want to use them, please reach out to the
+ * OpenZFS community, e.g., on GitHub or Slack.
+ */
+
 /*
  * Field manipulation macros for the drr_versioninfo field of the
  * send stream header.
  */
+#define	DMU_GET_STREAM_HDRTYPE(vi)	BF64_GET((vi), 0, 2)
+#define	DMU_SET_STREAM_HDRTYPE(vi, x)	BF64_SET((vi), 0, 2, x)
+
+#define	DMU_GET_FEATUREFLAGS(vi)	BF64_GET((vi), 2, 56)
+#define	DMU_SET_FEATUREFLAGS(vi, x)	BF64_SET((vi), 2, 56, x)
```

Comment:
With the second argument being a number of bits, I suppose here should be 54, not 56.

Reply:
Argh, I get those macros wrong every time. Thanks.